PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/DisableAutoCompleteCheck.java

```diff
@@ -0,0 +1,192 @@
+     */
+    static class DisableAutoCompleteVisitor extends JavaElementVisitor {
+
+        // Define constants for string literals
+        private static String SUGGESTION;
+        private static List<String> CLIENTS_TO_CHECK;
+        private static String METHOD_TO_CHECK;
+        private static final Logger LOGGER = Logger.getLogger(ServiceBusReceiverAsyncClientCheck.class.getName());
+
+        static {
+            try {
+                getRuleConfigData();
+            } catch (IOException e) {
+                LOGGER.log(Level.SEVERE, "Error loading client data", e);
+            }
+        }
+
+        // Define the holder for the problems found
+        private static ProblemsHolder holder;
+
+        /**
+         * Constructor for the visitor
+         *
+         * @param holder     The holder for the problems found
+         */
+        public DisableAutoCompleteVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+        }
+
+        /**
+         * This method is used to visit the declaration statements in the code.
+         * It checks for the declaration of Azure SDK ServiceBusReceiver & ServiceBusProcessor clients and
+         * whether the auto-complete feature is disabled.
+         * If the auto-complete feature is not disabled, a problem is registered with the ProblemsHolder.
+         *
+         * @param statement The declaration statement to visit
+         */
+        @Override
+        public void visitDeclarationStatement(PsiDeclarationStatement statement) {
+            super.visitDeclarationStatement(statement);
+
+            // Get the declared elements
+            PsiElement[] elements = statement.getDeclaredElements();
+
+            // Get the variable declaration
+            if (elements.length > 0 && elements[0] instanceof PsiVariable) {
+                PsiVariable variable = (PsiVariable) elements[0];
+
+                // Process the variable declaration
+                processVariableDeclaration(variable);
+            }
+        }
+
+        /**
+         * This method is used to process the variable declaration.
+         * It checks for the declaration of Azure SDK ServiceBusReceiver & ServiceBusProcessor clients and
+         * whether the auto-complete feature is disabled.
+         * If the auto-complete feature is not disabled, a problem is registered with the ProblemsHolder.
+         *
+         * @param variable The variable to process
+         */
+        private void processVariableDeclaration(PsiVariable variable) {
+
+            // Retrieve the client name (left side of the declaration)
+            PsiType clientType = variable.getType();
+
+            // Check the assignment part (right side)
+            PsiExpression initializer = variable.getInitializer();
+
+            // Check if the client type is an Azure SDK client
+            if (!clientType.getCanonicalText().startsWith("com.azure")) {
```

Comment:
Made a similar comment on other PR, to put this into util.

Reply:
yehp:)